core/src/main/java/com/microsoft/applicationinsights/channel/concrete/inprocess/InProcessTelemetryChannel.java

```diff
@@ -181,11 +210,42 @@ public void flush() {
         telemetryBuffer.flush();
     }
 
+    /**
+     * Sets the buffer size
+     * @param maxTelemetriesInBatch should be between MIN_NUMBER_OF_TELEMETRIES_PER_CONTAINER
+     *                              and MAX_NUMBER_OF_TELEMETRIES_PER_CONTAINER inclusive
+     */
+    public void setMaxTelemetriesInBatch(int maxTelemetriesInBatch) {
+        if (maxTelemetriesInBatch >= MIN_NUMBER_OF_TELEMETRIES_PER_CONTAINER &&
+                maxTelemetriesInBatch <= MAX_NUMBER_OF_TELEMETRIES_PER_CONTAINER) {
+            telemetryBuffer.setMaxTelemetriesInBatch(maxTelemetriesInBatch);
+        } else {
+            InternalLogger.INSTANCE.trace("Value %s is ignored, value should be between %s and %s, inclusive",
+                    maxTelemetriesInBatch, MIN_NUMBER_OF_TELEMETRIES_PER_CONTAINER, MAX_NUMBER_OF_TELEMETRIES_PER_CONTAINER);
+        }
+
+    }
+
+    /**
+     * Sets the time tow wait before flushing the internal buffer
+     * @param transmitBufferTimeoutInSeconds should be between MIN_TRANSMIT_BUFFER_TIMEOUT_IN_SECONDS
+     *                              and MAX_TRANSMIT_BUFFER_TIMEOUT_IN_SECONDS inclusive
+     */
+    public void setTransmitBufferTimeoutInSeconds(int transmitBufferTimeoutInSeconds) {
+        if (transmitBufferTimeoutInSeconds >= MIN_TRANSMIT_BUFFER_TIMEOUT_IN_SECONDS &&
```

Comment:
Same comment as above regarding the behavior in case the provided value is higher/lower than the limits.